src/Abp/EntityHistory/EntitySnapshotManagerBase.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Abp.Dependency;
+using Abp.Domain.Entities;
+using Abp.Domain.Repositories;
+using Abp.Json;
+using Abp.Linq;
+
+namespace Abp.EntityHistory
+{
+    public abstract class EntitySnapshotManagerBase : IEntitySnapshotManager, ITransientDependency
+    {
+        protected readonly IRepository<EntityChange, long> EntityChangeRepository;
+        private readonly IAsyncQueryableExecuter _asyncQueryableExecuter;
```

Comment:
we use property injection for `IAsyncQueryableExecuter`

https://github.com/aspnetboilerplate/aspnetboilerplate/blob/e0ded5d8702f389aa1f5947d3446f16aec845287/src/Abp/Application/Services/AsyncCrudAppService.cs#L96-L102